[Decomposition] unbind

salilsdesai opened the pull request:
Summary: Copy decomp from caffe2/torch/_refs/__init__.py and include it in core_aten_decompositions

Test Plan: OSS + Phabricator Tests

Differential Revision: D48871742




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @Xia-Weiwen @ngimel

SherlockNoMad commented on torch/_decomp/__init__.py:
```diff
@@ -368,6 +368,7 @@ def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
             aten.tril_,
             aten.triu,
             aten.triu_,
+            aten.unbind,
```
Inductor has lowering for this: implying inductor would want this op preserved in graph. 
Please exclude this from inductor decomp. 

see _inductor/lowering.py
and _inductor/decomposition.py

larryliu0820 replied:
I imagine inductor CI will break on this PR, right?

SherlockNoMad reviewed (approved):
Remember to export the new commits to OSS.

salilsdesai committed:
[Decomposition] unbind (#108546)

Summary:
bypass-github-pytorch-ci-checks


Decomp already exists. Include it in core_aten_decompositions

Decomp:
https://www.internalfb.com/code/fbsource/[e69bf00ff87a55c9a30bd7905881661ff05fa211]/xplat/caffe2/torch/_refs/__init__.py?lines=3810

Test Plan: OSS + Phabricator Tests

Reviewed By: SS-JIA

Differential Revision: D48871742
